cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go

```diff
@@ -297,6 +297,174 @@ func (s *Server) GetAvailablePackageVersions(ctx context.Context, request *corev
+
+	// https://carvel.dev/kapp-controller/docs/latest/packaging/#package-metadata
+	name, found, err := unstructured.NestedString(pkgMeta.Object, "metadata", "name")
+	if err != nil || !found || name == "" {
+		return nil, status.Errorf(codes.Internal, "required field metadata.name not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	namespace, found, err := unstructured.NestedString(pkgMeta.Object, "metadata", "namespace")
+	if err != nil || !found {
+		return nil, status.Errorf(codes.Internal, "required field metadata.namespace not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	detail.Name = name
+	displayName, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "displayName")
+	if err != nil || !found {
+		return nil, status.Errorf(codes.Internal, "required field spec.displayName not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	detail.DisplayName = displayName
+
+	// Carvel uses base64-encoded SVG data for IconSVGBase64, whereas we need
+	// a url, so convert to a data-url.
+	iconSVGBase64, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "iconSVGBase64")
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "required field spec.iconSVGBase64 not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	if found && iconSVGBase64 != "" {
+		detail.IconUrl = fmt.Sprintf("data:image/svg+xml;base64,%s", iconSVGBase64)
+	}
+
+	shortDescription, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "shortDescription")
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "required field spec.shortDescription not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	if found {
+		detail.ShortDescription = shortDescription
+	}
+
+	categories, found, err := unstructured.NestedStringSlice(pkgMeta.Object, "spec", "categories")
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "required field spec.categories not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	if found {
+		detail.Categories = categories
+	}
+
+	detail.Version = &corev1.PackageAppVersion{
+		PkgVersion: version,
+	}
+
+	longDescription, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "longDescription")
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "required field spec.longDescription not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	if found {
+		detail.LongDescription = longDescription
+	}
+	detail.Readme = longDescription
+
+	supportDescription, found, err := unstructured.NestedString(pkgMeta.Object, "spec", "supportDescription")
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "required field spec.supportDescription not found on kapp-controller packagemetadata: %v\n%v", err, pkgMeta.Object)
+	}
+	if found {
+		detail.Readme = detail.Readme + "\n\n" + supportDescription
```

Comment:
Not sure, but instead of just concatenating here, have you tried using some markdown (like creating two sections or so) to make it look less sparse ? Perhaps the final rendered result is worse, haha, not sure.

Reply:
Yes, I did consider it, but given that there's no standard for the format of carvel long description or support description yet, I didn't want to add any markdown. We can consider it in the future once we've some actual examples of carvel packages (interested to see the bitnami ones).